src/server/journal/streamer.cc

```diff
@@ -41,7 +41,9 @@ JournalStreamer::JournalStreamer(journal::Journal* journal, Context* cntx)
 }
 
 JournalStreamer::~JournalStreamer() {
-  DCHECK_EQ(in_flight_bytes_, 0u);
+  if (!cntx_->IsCancelled()) {
+    DCHECK_EQ(in_flight_bytes_, 0u);
```

Comment:
How did we not trigger this before ? Or did we just deadlocked because `WaitForInFlightToCOmplete() would never progress` ?

Reply:
idk why we didn't trigger this before, but indeed this dead locks